### PR TITLE
ARROW-17973: [C++] Expression::ToString wrong for nullary function call

### PR DESCRIPTION
**Pull request by sanjibansg**
This PR fixes the `ToString()` method in the `Expression` class to handle functions with no arguments.

### CODE REVIEW

**Review by pitrou — CHANGES_REQUESTED** [review]
Thanks a lot @sanjibansg . Just one suggestion below.

File: cpp/src/arrow/compute/exec/expression_test.cc [hunk 1]
```diff
@@ -284,8 +284,7 @@ TEST(Expression, ToString) {
       "allow_time_overflow=false, allow_decimal_truncate=false, "
       "allow_float_truncate=false, allow_invalid_utf8=false})");
 
-  // NB: corrupted for nullary functions but we don't have any of those
-  EXPECT_EQ(call("widgetify", {}).ToString(), "widgetif)");
+  EXPECT_EQ(call("widgetify", {}).ToString(), "widgetify()");
```
[the remark this answers is not in the document]
Comment (sanjibansg) [reply cpp/src/arrow/compute/exec/expression_test.cc:287]: added a test with `RoundOptions()`

[the remark this answers is not in the document]
Comment (pitrou) [reply cpp/src/arrow/compute/exec/expression_test.cc:287]: Ok, but I meant a test for a nullary function call...

[the remark this answers is not in the document]
Comment (sanjibansg) [reply cpp/src/arrow/compute/exec/expression_test.cc:287]: added a test with `RandomOptions()`

**Review by pitrou — APPROVED** [review]

**Pull request merged**